### PR TITLE
a better Error type when conversion from Date to time::Date doesn't work

### PR DESCRIPTION
**Pull request by eeeebbbbrrrr**

### CODE REVIEW

File: pgx/src/datum/date.rs [hunk 1]
```diff
@@ -98,24 +98,54 @@ impl Date {
...
-    fn from(date: time::Date) -> Self {
-        Date::from_pg_epoch_days(date.to_julian_day() - POSTGRES_EPOCH_JDATE)
+mod with_time_crate {
+    use crate::{Date, POSTGRES_EPOCH_JDATE};
+    use core::fmt::{Display, Formatter};
+    use std::error::Error;
+
+    #[derive(Debug, PartialEq, Clone)]
+    #[non_exhaustive]
+    pub struct TryFromDateError(pub Date);
```
Comment (thomcc) [inline pgx/src/datum/date.rs:108]: Do we want to export this in a way users can access it?

**Review by thomcc — APPROVED** [review]
It looks mostly okay to me, but the error type probably should be exported unless we have a reason not to.

File: pgx/src/datum/date.rs (hunk 1 above)
Comment (eeeebbbbrrrr) [reply pgx/src/datum/date.rs:108]: I don’t see that they’ll need to reference it by name.

Comment (eeeebbbbrrrr) [reply pgx/src/datum/date.rs:108]: It’s under this time-crate feature flag anyways. And honestly, I hope that’ll go away soon

Comment (eeeebbbbrrrr) [reply pgx/src/datum/date.rs:111]: alrighty.  done

**Pull request merged**